src/jit/codegenlegacy.cpp

```diff
@@ -21307,6 +21310,21 @@ regNumber CodeGen::genPInvokeCallProlog(LclVarDsc*            frameListRoot,
 
     noway_assert(compiler->lvaInlinedPInvokeFrameVar != BAD_VAR_NUM);
 
+#ifdef _ARM_
+    if (compiler->opts.ShouldUsePInvokeHelpers())
+    {
+        regNumber baseReg;
+        int       adr = compiler->lvaFrameAddress(compiler->lvaInlinedPInvokeFrameVar, true, &baseReg, 0);
```

Comment:
I suspect that the argument registers are filled with the arguments to be passed to the PInvoke at this point, and this will overwrite them. Could you please manually inspect the generated code for correctness?